server/scsynth/SC_PortAudio.cpp

```diff
@@ -278,18 +278,28 @@ bool SC_PortAudioDriver::DriverSetup(int* outNumSamples, double* outSampleRate)
                 pdi->maxInputChannels, pdi->maxOutputChannels);
     }
 
-    mDeviceInOut[0] = GetPaDeviceFromName(mWorld->hw->mInDeviceName, true);
-    mDeviceInOut[1] = GetPaDeviceFromName(mWorld->hw->mOutDeviceName, false);
+    auto m_inDeviceName = mWorld->hw->mInDeviceName;
+    auto m_outDeviceName = mWorld->hw->mOutDeviceName;
+    mDeviceInOut[0] = GetPaDeviceFromName(m_inDeviceName, true);
+    mDeviceInOut[1] = GetPaDeviceFromName(m_outDeviceName, false);
 
     // report requested devices
     fprintf(stdout, "\nRequested devices:\n");
     if (mWorld->mNumInputs) {
-        fprintf(stdout, "  In (matching device %sfound):\n  - %s\n", (mDeviceInOut[0] == paNoDevice ? "NOT " : ""),
-                mWorld->hw->mInDeviceName);
+        auto nameIsEmpty = (m_inDeviceName && !m_inDeviceName[0]) || (m_inDeviceName == nullptr);
```

Comment:
this is suspicious to me.. is the null case where the user doesn't supply the option, and the empty string case when the user passes `""`? if so we should just combine both cases to empty string -- i see code elsewhere in the project (https://github.com/supercollider/supercollider/blob/develop/server/scsynth/SC_CoreAudio.cpp#L600) that doesn't make this distinction. i can fix this, it would be a good time for https://github.com/supercollider/supercollider/issues/3117.

Reply:
Yes, AFAIU it's for not passing any device and for passing `""`. 

if #3117 is addressed, would that mean that `mWorld->hw->mInDeviceName` becomes `std::string`? Of so, this PR would need to wait for that change to be made first, right?

Reply:
> would that mean that mWorld->hw->mInDeviceName becomes std::string? Of so, this PR would need to wait for that change to be made first, right?

yes, probably it would become a std::string. but no, i don't know when i would want to fix that, so we don't need to wait :)